processing/src/main/java/org/apache/druid/segment/serde/NullColumnPartSerde.java

```diff
@@ -0,0 +1,262 @@
+import org.apache.druid.segment.column.BitmapIndex;
+import org.apache.druid.segment.column.BitmapIndexes;
+import org.apache.druid.segment.column.DictionaryEncodedColumn;
+import org.apache.druid.segment.column.NumericColumn;
+import org.apache.druid.segment.data.BitmapSerdeFactory;
+import org.apache.druid.segment.data.IndexedInts;
+import org.apache.druid.segment.data.ReadableOffset;
+import org.apache.druid.segment.vector.MultiValueDimensionVectorSelector;
+import org.apache.druid.segment.vector.NilVectorSelector;
+import org.apache.druid.segment.vector.ReadableVectorOffset;
+import org.apache.druid.segment.vector.SingleValueDimensionVectorSelector;
+import org.apache.druid.segment.vector.VectorObjectSelector;
+import org.apache.druid.segment.vector.VectorValueSelector;
+
+import javax.annotation.Nullable;
+import java.nio.channels.WritableByteChannel;
+import java.util.Objects;
+
+/**
+ * A ColumnPartSerde to read and write null-only columns.
+ * Its serializer is no-op as nothing is stored for null-only columns.
+ * Its deserializer creates necessary column metadata and indexes when the column is read.
+ */
+public class NullColumnPartSerde implements ColumnPartSerde
+{
+  private static final Serializer NOOP_SERIALIZER = new Serializer()
+  {
+    @Override
+    public long getSerializedSize()
+    {
+      return 0;
+    }
+
+    @Override
+    public void writeTo(WritableByteChannel channel, FileSmoosher smoosher)
+    {
+    }
+  };
+
+  private final int numRows;
+  private final BitmapSerdeFactory bitmapSerdeFactory;
+  private final NullNumericColumn nullNumericColumn;
+  private final NullDictionaryEncodedColumn nullDictionaryEncodedColumn;
+  private final BitmapIndex bitmapIndex;
+
+  @JsonCreator
+  public NullColumnPartSerde(
+      @JsonProperty("numRows") int numRows,
+      @JsonProperty("bitmapSerdeFactory") BitmapSerdeFactory bitmapSerdeFactory
+  )
+  {
+    this.numRows = numRows;
+    this.bitmapSerdeFactory = bitmapSerdeFactory;
+    this.nullNumericColumn = new NullNumericColumn();
+    this.nullDictionaryEncodedColumn = new NullDictionaryEncodedColumn();
+    this.bitmapIndex = BitmapIndexes.forNilColumn(() -> numRows, bitmapSerdeFactory.getBitmapFactory());
+  }
+
+  @JsonProperty
+  public int getNumRows()
+  {
+    return numRows;
+  }
+
+  @JsonProperty
+  public BitmapSerdeFactory getBitmapSerdeFactory()
+  {
+    return bitmapSerdeFactory;
+  }
+
+  @Nullable
+  @Override
+  public Serializer getSerializer()
+  {
+    return NOOP_SERIALIZER;
+  }
+
+  @Override
+  public Deserializer getDeserializer()
+  {
+    return (buffer, builder, columnConfig) -> builder
+        .setHasMultipleValues(false)
+        .setHasNulls(true)
+        .setFilterable(true)
+        .setBitmapIndex(Suppliers.ofInstance(bitmapIndex))
+        .setNumericColumnSupplier(Suppliers.ofInstance(nullNumericColumn))
+        .setDictionaryEncodedColumnSupplier(Suppliers.ofInstance(nullDictionaryEncodedColumn));
```

Comment:
this isn't correct, both of these statements both set the column supplier in the builder, I think you probably only need the dictionary encoded column supplier, which i think should behave correctly, and then can drop the `NullNumericColumn`?

Also, does this column have no type?

Reply:
Ah good catch. I planned to fix it but forgot until you pointed it out. I removed `NullNumericColumn`. The column type is set by `ColumnDescriptor`.